jenkins/jjb/papr.yaml

```diff
@@ -0,0 +1,132 @@
+- job-template:
+    name: 'papr-trigger-{ghowner}-{ghrepo}'
+    description: |
+      This job is triggered on both branch push and PR events for the
+      {ghowner}/{ghrepo} GitHub repo. It then converts those events to
+      Kubernetes jobs. See https://github.com/projectatomic/paci.
+    concurrent: false
+    node: master
+    wrappers:
+      - timestamps
+      - ansicolor
+    # for manual triggering
+    parameters:
+      - string:
+          name: PAPR_GITHUB_BRANCH
+          description: >
+            Branch to test (incompatible with PAPR_GITHUB_PULL_ID).
+      - string:
+          name: PAPR_GITHUB_PULL_ID
+          description: >
+            Pull request ID to test (incompatible with PAPR_GITHUB_BRANCH).
+      - string:
+          name: PAPR_TESTSUITES
+          description: >
+            Pipe-separated list of contexts. only the testsuites which
+            set these contexts will be run. Leave blank to run all the
+            testsuites.
+    scm:
+      # NB: always place first to take the index 0 in GIT_ env vars
+      - git:
+          url: 'https://github.com/{ghowner}/{ghrepo}'
+          # we also fetch pull refs so GHPRB can do a checkout
+          refspec: >
+            +refs/heads/*:refs/remotes/origin/*
+            +refs/pull/*:refs/remotes/origin/pr/*
+          basedir: checkout
+          # we only want the SCM plugin to handle regular branches, PRs are
+          # handled by GHPRB (the default is '**', which is all refs)
+          branches: ['origin/*']
+          # we don't actually need the full history, so minimize bandwidth
+          # (but still do a checkout to check for .papr.yml)
+          do-not-fetch-tags: true
+          # https://issues.jenkins-ci.org/browse/JENKINS-44323
+          #shallow-clone: true
+          wipe-workspace: false
+    properties:
+      - github: # required by GHPRB
+          url: 'https://github.com/{ghowner}/{ghrepo}'
+    triggers:
+      - github # branch pushes
+      - github-pull-request:
+
+          github-hooks: true
+          auth-id: '7a64217f-077c-4fde-a9a3-fd0ae37d4e0a'
```

Comment:
Should this be public?

Reply:
Yeah, it's just an object ID, not a credential in any way. Humans can customize those (like I did for `github-webhook-shared-secret`) but I just used what Jenkins auto-generated.